cps/kni.c

```diff
@@ -721,20 +777,24 @@ rd_modroute(const struct nlmsghdr *req, void *arg)
 	switch(rm->rtm_family) {
 	case AF_INET:
 		mnl_attr_parse(req, sizeof(*rm), data_ipv4_attr_cb, tb);
-		attr_get(update, rm->rtm_family, tb);
+		err = attr_get(update, rm->rtm_family, tb);
 		break;
 	case AF_INET6:
 		mnl_attr_parse(req, sizeof(*rm), data_ipv6_attr_cb, tb);
-		attr_get(update, rm->rtm_family, tb);
+		err = attr_get(update, rm->rtm_family, tb);
 		break;
 	default:
 		CPS_LOG(NOTICE, "Unrecognized family in netlink event: %u\n",
 			rm->rtm_family);
+		err = -EAFNOSUPPORT;
 		break;
 	}
 
 	if (update->valid)
 		updates->count++;
+out:
+	if ((req->nlmsg_flags & NLM_F_ACK) || err)
+		rd_send_err(req, err);
```

Comment:
At this point, we have not updated the LPM table, so an error can still happen. We should send the acknowledgment or error reply later in the code.